api/src/test/java/org/openmrs/module/htmlformentryui/tag/UiIcludeTagHandlerTest.java

```diff
@@ -0,0 +1,107 @@
+package org.openmrs.module.htmlformentryui.tag;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openmrs.Patient;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.w3c.dom.DOMException;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+import org.apache.xerces.impl.xs.opti.DefaultNode;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Context.class)
+public class UiIcludeTagHandlerTest {
+	
+	private FormEntrySession session;
+	private UiIncludeTagHandler handler;
+	private static final String PATIENT_UUID = "ea7ae6ac-e9de-430f-a5ce-4adcf3b1ed43";
+	private Node node;
+	
+	@Before
+	public void setup() throws Exception {
+		mockStatic(Context.class);
+		when(Context.getAuthenticatedUser()).thenReturn(new User(1));
+		Patient pat = new Patient();
+		pat.setUuid(PATIENT_UUID);
+		session = new FormEntrySession(pat, "xml", null);
+		handler = spy(new UiIncludeTagHandler());
+		node = new NodeTest();
+		doReturn("path/page?age=5").when(handler).getAttribute(node, "fragment", null);
+		doReturn("retired=true&patientId=$patient.uuid").when(handler).getAttribute(node, "fragmentParams", null);
+	}
+	
+	@Test
+	public void doIncludeFragment_shouldParseUrlParameter() {
+		handler.doIncludeFragment(node, null, session, null, null);
+		//TODO: Assert expected behaviour
```

Comment:
Looks like the assertions still need to be added here?

Reply:
True that. I'm working on the assertions just like the `TODO` above states.